### ISSUE TITLE
Leveraging new (jekyll 3) `layout` metadata when `page` front-matter not present

### ISSUE THREAD

**Issue by k-funk** [jekyll/jekyll-seo-tag#108]
Sometimes I use layout front-matter/metadata instead of page front-matter when I have multiple pages that have mostly the same content. However, since Jekyll 3 no longer merges the layout metadata with page front-matter, `title` and `description` on a layout don't get populated by jekyll seo tag. What do you think about:

```
{% if page.title || layout.title %}
  {% assign seo_title = page.title || layout.title %}
  ...
{% endif %}
```

For context, I create api docs with urls/pages like:
`/docs/api/ruby.html`
`/docs/api/python.html`
`/docs/api/java.html`
`/docs/api/node.html`
`/docs/api/...`
All those pages use the same layout, and should have the same title, description, and corresponding metadata.

**Comment by pathawks** [comment]
Could you describe the problem you are having in a bit more detail?

Google [specifically recommends against boilerplate titles](https://support.google.com/webmasters/answer/35624?hl=en#3), so I am initially opposed to this idea, but maybe I am just not fully understanding the problem you are trying to solve. Can you give some examples of what you would expect the `<title>` to be on some of the URLs listed above? Wouldn't you want: `Ruby API Docs`, `Python API Docs`, etc...?

**Comment by k-funk** [comment]
I s'pose the "problem" is mostly that of having the same content in multiple places, and that doesn't feel very DRY. `Ruby API Docs`, `Python API Docs`... may be what I go with, but in my mind they are all the same page with a different context. The only thing that is different within the page is the code samples.

When I've worked with web applications in the past, I've used the query params (`/docs/api?lang=python`) to get the associated code samples, which seems the most correct, so the title remained the same.
